Deduplicate some copied event conversion code between WebCore and WebKit2

hortont424 opened the pull request:
#### e3ece76059110c8d76f28c9e8c143bf20a3051e3
<pre>
Deduplicate some copied event conversion code between WebCore and WebKit2
<a href="https://bugs.webkit.org/show_bug.cgi?id=291097">https://bugs.webkit.org/show_bug.cgi?id=291097</a>
<a href="https://rdar.apple.com/148613835">rdar://148613835</a>

Reviewed by Abrar Rahman Protyasha.

A long time ago, a bunch of event conversion code (especially code to read
properties off of NSEvent) was copied from WebCore to WebKit.

De-duplicate that copy-pasting, propagating changes that were made in only
one place since the divergence.

Add some helpers for converting between WebCore and WebKit enums to make
this sharing easier (mostly factored out of existing places).

Adopt the new helpers in a few other random places that had duplicative code.

* Source/WebCore/PAL/pal/spi/mac/NSEventSPI.h:
* Source/WebCore/platform/mac/PlatformEventFactoryMac.h:
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::globalPointForEvent):
(WebCore::pointForEvent):
(WebCore::currentMouseButton):
(WebCore::mouseButtonForEvent):
(WebCore::currentlyPressedMouseButtons):
(WebCore::mouseEventTypeForEvent):
(WebCore::clickCountForEvent):
(WebCore::textFromEvent):
(WebCore::unmodifiedTextFromEvent):
(WebCore::isKeypadEvent):
(WebCore::isKeyUpEvent):
(WebCore::modifiersForEvent):
(WebCore::modifiersForModifierFlags):
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::platform):
(WebKit::kit):
(WebKit::forceForEvent):
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
(WebKit::WebKit2PlatformWheelEvent::WebKit2PlatformWheelEvent):
(WebKit::WebKit2PlatformKeyboardEvent::WebKit2PlatformKeyboardEvent):
(WebKit::WebKit2PlatformTouchEvent::WebKit2PlatformTouchEvent):
(WebKit::WebKit2PlatformGestureEvent::WebKit2PlatformGestureEvent):
* Source/WebKit/Shared/WebEventConversion.h:
* Source/WebKit/Shared/WebMouseEvent.cpp:
(WebKit::mouseButton):
* Source/WebKit/Shared/mac/WebEventFactory.h:
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::WebEventFactory::createWebMouseEvent):
(WebKit::WebEventFactory::createWebWheelEvent):
(WebKit::WebEventFactory::createWebKeyboardEvent):
(WebKit::currentMouseButton): Deleted.
(WebKit::mouseButtonForEvent): Deleted.
(WebKit::currentlyPressedMouseButtons): Deleted.
(WebKit::mouseEventTypeForEvent): Deleted.
(WebKit::clickCountForEvent): Deleted.
(WebKit::globalPointForEvent): Deleted.
(WebKit::pointForEvent): Deleted.
(WebKit::textFromEvent): Deleted.
(WebKit::unmodifiedTextFromEvent): Deleted.
(WebKit::isKeypadEvent): Deleted.
(WebKit::isKeyUpEvent): Deleted.
(WebKit::WebEventFactory::webEventModifiersForNSEventModifierFlags): Deleted.
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):
(WebKit::WebAutomationSession::platformWebModifiersFromRaw):

Canonical link: <a href="https://commits.webkit.org/293379@main">https://commits.webkit.org/293379@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7b5376458b69a42e2ef6908d1aa098e9aa772cba

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98382 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18013 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8244 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103503 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/48910 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/100426 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18306 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26465 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74890 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32054 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101386 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13884 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/88855 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55249 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13664 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/6820 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48352 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/83621 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6897 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/105876 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25470 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/18545 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83871 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25846 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85056 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83346 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21123 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27990 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/5662 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19122 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25429 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/30606 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25247 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/28567 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26822 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/platform/mac/PlatformEventFactoryMac.h:
```diff
@@ -45,17 +45,30 @@ class PlatformEventFactory {
 // FIXME: This function doesn't really belong in this header.
 WEBCORE_EXPORT NSPoint globalPoint(const NSPoint& windowPoint, NSWindow *);
 
-// FIXME: WebKit2 has a lot of code copied and pasted from PlatformEventFactoryMac in WebEventFactory. More of it should be shared with WebCore.
```
This FIXME always irked me, so thanks!

aprotyas commented on Source/WebKit/Shared/WebEventConversion.cpp:
```diff
@@ -38,6 +38,154 @@
+#endif
+
+    default:
+        RELEASE_ASSERT_NOT_REACHED();
```
Maybe just keep `ASSERT_NOT_REACHED()`, like before? Seems harsh to crash here

aprotyas replied:
Having said that we were inconsistent before too, I see at least one `RELEASE_ASSERT_NOT_REACHED()` in the deleted lines.

hortont424 committed:
Deduplicate some copied event conversion code between WebCore and WebKit2
https://bugs.webkit.org/show_bug.cgi?id=291097
rdar://148613835

Reviewed by Abrar Rahman Protyasha.

A long time ago, a bunch of event conversion code (especially code to read
properties off of NSEvent) was copied from WebCore to WebKit.

De-duplicate that copy-pasting, propagating changes that were made in only
one place since the divergence.

Add some helpers for converting between WebCore and WebKit enums to make
this sharing easier (mostly factored out of existing places).

Adopt the new helpers in a few other random places that had duplicative code.

* Source/WebCore/PAL/pal/spi/mac/NSEventSPI.h:
* Source/WebCore/platform/mac/PlatformEventFactoryMac.h:
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::globalPointForEvent):
(WebCore::pointForEvent):
(WebCore::currentMouseButton):
(WebCore::mouseButtonForEvent):
(WebCore::currentlyPressedMouseButtons):
(WebCore::mouseEventTypeForEvent):
(WebCore::clickCountForEvent):
(WebCore::textFromEvent):
(WebCore::unmodifiedTextFromEvent):
(WebCore::isKeypadEvent):
(WebCore::isKeyUpEvent):
(WebCore::modifiersForEvent):
(WebCore::modifiersForModifierFlags):
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::platform):
(WebKit::kit):
(WebKit::forceForEvent):
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
(WebKit::WebKit2PlatformWheelEvent::WebKit2PlatformWheelEvent):
(WebKit::WebKit2PlatformKeyboardEvent::WebKit2PlatformKeyboardEvent):
(WebKit::WebKit2PlatformTouchEvent::WebKit2PlatformTouchEvent):
(WebKit::WebKit2PlatformGestureEvent::WebKit2PlatformGestureEvent):
* Source/WebKit/Shared/WebEventConversion.h:
* Source/WebKit/Shared/WebMouseEvent.cpp:
(WebKit::mouseButton):
* Source/WebKit/Shared/mac/WebEventFactory.h:
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::WebEventFactory::createWebMouseEvent):
(WebKit::WebEventFactory::createWebWheelEvent):
(WebKit::WebEventFactory::createWebKeyboardEvent):
(WebKit::currentMouseButton): Deleted.
(WebKit::mouseButtonForEvent): Deleted.
(WebKit::currentlyPressedMouseButtons): Deleted.
(WebKit::mouseEventTypeForEvent): Deleted.
(WebKit::clickCountForEvent): Deleted.
(WebKit::globalPointForEvent): Deleted.
(WebKit::pointForEvent): Deleted.
(WebKit::textFromEvent): Deleted.
(WebKit::unmodifiedTextFromEvent): Deleted.
(WebKit::isKeypadEvent): Deleted.
(WebKit::isKeyUpEvent): Deleted.
(WebKit::WebEventFactory::webEventModifiersForNSEventModifierFlags): Deleted.
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):
(WebKit::WebAutomationSession::platformWebModifiersFromRaw):

Canonical link: https://commits.webkit.org/293379@main
